mava/systems/tf/mappo/training.py

```diff
@@ -414,32 +507,42 @@ def _backward(self) -> None:
             agent_key = self._agent_net_keys[agent]
 
             # Get trainable variables.
-            policy_variables = (
-                self._observation_networks[agent_key].trainable_variables
-                + self._policy_networks[agent_key].trainable_variables
-            )
-            critic_variables = self._critic_networks[agent_key].trainable_variables
+            if self._use_single_optimizer:
+                policy_variables = (
```

Comment:
I see why we are keeping the name `policy_variables` here to work with the code further downstream, but we should be careful with being too loose with our code. This is clearly not just the policy variables, so the name can be confusing.

Reply:
This also applies further down steam, i.e. when we compute the gradients it will also not only be the policy gradients.

Reply:
Updated this to be more explicit. Let me know what you think?

Reply:
Yeah, I think this is better. Thanks.